### PR TITLE
Refactor Config.For function

### PR DESCRIPTION
**Pull request by liornoy**
This commit separates the logic of creating the Config
struct in the For function into 3 different functions.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -170,44 +170,68 @@ func For(resources ClusterResources, validate Validate) (*Config, error) {
...
+
+	cfg.Pools, err = parsePools(resources)
+	if err != nil {
+		return nil, err
 	}
 
+	return cfg, nil
+}
+
+func parseBFDProfiles(resources ClusterResources) (map[string]*BFDProfile, error) {
```
Comment (fedepaol) [inline internal/config/config.go:193]: I'd reuse the same "for*" pattern in naming

Comment (liornoy) [reply internal/config/config.go]: done

File: internal/config/config.go [hunk 2]
```diff
@@ -170,44 +170,68 @@ func For(resources ClusterResources, validate Validate) (*Config, error) {
...
+
+	cfg.Pools, err = forPools(resources)
+	if err != nil {
+		return nil, err
 	}
 
+	return cfg, nil
+}
+
+func forBFDProfiles(resources ClusterResources) (map[string]*BFDProfile, error) {
```
Comment (fedepaol) [inline internal/config/config.go:193]: I'd move this (and the others) as `bfdProfilesFor`, sorry I was not clear with my previous comment

Comment (liornoy) [reply internal/config/config.go]: np, fixed.

**Comment by fedepaol** [comment]
LGTM

**Pull request merged**